Kostin danila

kstdnl opened the pull request.

kstdnl committed:
5.4

kstdnl committed:
Rename 5.4 to 5.4.py

kstdnl committed:
6.3

kstdnl committed:
6.1

Nastasia8 commented on 5/5.4.py:
```diff
@@ -0,0 +1,52 @@
+ 
+ 
+main()
+ 
```
Зашибись сколько комментариев, как Вас не разорвало...

kstdnl replied:
скажите, пожалуйста, какие строки прокомментировать, может убежу все таки.

Nastasia8 commented on 6/6.3.py:
```diff
@@ -0,0 +1,48 @@
+n = int(input())
+numbers = list(map(int, input().split()))
+heap = build(numbers)
+res = deque()
```
Вот неужели так трудно включить мозг? Разве без deque() задачу нельзя было решить?
От Вас, конечно, такого не ожидала.

kstdnl replied:
мозг включён, как и обычно, конечно можно было и не использовать deque, но когда спрашивал у других ребят насчёт ошибки в коде, который написал, мне посоветовали попробовать это. Обидно конечно, но стараюсь держаться :)

kstdnl committed:
5.6.py

Во избежание грубости по отношению ко мне, комментирую все добавленные строкив коде 5.6 к коду 5.4, а именно измененную функцию main(). Функция update(...) была скопирована с учебного кода, написанного на уроке.

kstdnl committed:
6.2.py

суть задачи 6.2 - реализовать приоритетную очередь, поддерживающую три операции:  добавить элемент, извлечь максимальный элемент, удалить элемент с заданным индексом из очереди. С первой задачей справляется старая функция - add(непосредственно связанная с методом shift up), для второй задачи описан та же старая функция get_max, третья задача, новая, решается благодаря функции direct_extract, принцип которой основан на тех же методах shift ... , удаляет элементы с помощью встроенной функции pop(строка 35). Если последний элемент в куче меньше удаляемого элемента, то опускаем элемент, иначе поднимаем. По итогу работы функции возвращаем i.
В функции main() дополнительно описываем 3 запрос(запрос удаления элемента). Если первый символ в введенной строке - тройка, то если элемент с таким индексом существовал, то выводим значение удалённого элемента, иначе, если элемента с таким индексом не существует выводим -1

kstdnl commented:
Добавил задачи 5.4 и 6.2 С КОММЕНТАРИЯМИ)
